include/abb_librws/rws_rapid.h

```diff
@@ -371,6 +372,243 @@ struct RAPIDRecord : public RAPIDSymbolDataAbstract
+   * \param array_size specifying the size of the array to allocate
+   * \param value initiale value of all the item of the array
+   */
+  RAPIDArray(
+    const std::string& array_type_name, 
+    const size_t array_size,
+    const value_type& value = value_type()
+  )
+  :   array_type_name_ { array_type_name }
+  ,   container_ {std::vector<value_type>(array_size, value)}
+  {}
+
+  /**
+   * \brief A constructor that takes an array to initialize the underlying container
+   * 
+   * \param array_type_name specifying the name of the RAPID type of the contained items
+   * \param array Array that will be used to initialize the container
+   */
+  RAPIDArray(const std::string& array_type_name, const std::vector<value_type>&& array)
+  :   array_type_name_ {array_type_name}
+  ,   container_ {array}
+  {}
+
+  /**
+   * \brief A method for constructing a RAPID symbol data value string.
+   *
+   * \return std::string containing the constructed string.
+   */
+  std::string constructString() const override
+  {
+    std::stringstream ss;
+
+    ss << "[";
+
+    for (size_t i = 0; i < container_.size(); ++i)
+    {
+      ss << container_.at(i).constructString();
+
+      if (i != container_.size() - 1)
+      {
+        ss << ",";
+      }
+    }
+
+    ss << "]";
+
+    return ss.str();
+  }
+
+  /**
+   * \brief A method for parsing a RAPID symbol data value string.
+   *
+   * \param value_string containing the string to parse.
+   */
+  void parseString(const std::string& value_string) override
+  {
+    std::vector<std::string> substrings = extractDelimitedSubstrings(value_string);
+
+    if(container_.size() == substrings.size())
+    {
+      for (size_t i = 0; i < substrings.size(); ++i)
+      {
+        container_.at(i).parseString(substrings.at(i));
+      }
+    }
+  }
+
+  /**
+   * \brief A method for getting the type of item stored in the RAPID array.
+   *
+   * \return std::string containing the type.
+   */
+  std::string getType() const override
+  {
+    return array_type_name_;
+  }
+
+  /**
+   * \brief Operator for copying the RAPID array to another RAPID array.
+   *
+   * \param other containing the RAPID array to copy.
+   *
+   * \return RAPIDArray& containing the copy.
+   */
+  RAPIDArray<value_type>& operator=(const RAPIDArray<value_type>& other)
+  {
+    if (this != &other)
+    {
+      if (array_type_name_ == other.array_type_name_)
+      {
+        if (container_.size() == other.container_.size())
+        {
+          for (size_t i = 0; i < container_.size(); ++i)
+          {
+            container_.at(i).parseString(other.container_.at(i).constructString());
+          }
+        }
+      }
+    }
+
+    return *this;
+  }
+
+  value_type& operator[](const size_t index) {
+    ensureIndexIsValid(index);
+    return container_[index];
+  }
+
+  value_type const& at(const size_t index) const {
+    ensureIndexIsValid(index);
+    return container_.at(index);
+  }
```

Comment:
I suggest inheriting `RAPIDArray` from the container type and have all container operations for free. This is a proper case for using multiple inheritance ([LSS](https://en.wikipedia.org/wiki/Liskov_substitution_principle) should hold).